### PR TITLE
Evaluate lock contention areas

### LINKED ISSUE #1543: Re-evaluate places where we are taking locks across await points
## Description

We need to evaluate the places where we are calling Wallet::data and data_mut and holding the lock across await points

## Motivation

Possible risk of dead locks.

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

There are some places where we need to be very cautious about holding locks. I have done some small refactoring to try to ensure that locks do not occur, and evaluated the places where locks seem likely.

## Links to any relevant issues

Closes #1543

### CODE REVIEW

File: sdk/src/wallet/core/mod.rs [hunk 1]
```diff
@@ -94,7 +94,7 @@ pub struct WalletInner<S: SecretManage = SecretManager> {
     #[cfg(feature = "storage")]
     pub(crate) storage_options: StorageOptions,
     #[cfg(feature = "storage")]
-    pub(crate) storage_manager: tokio::sync::RwLock<StorageManager>,
+    pub(crate) storage_manager: StorageManager,
```
Comment (thibault-martinez) [inline sdk/src/wallet/core/mod.rs:100]: Why is it not needed anymore?

File: sdk/src/wallet/core/operations/stronghold_backup/mod.rs [hunk 2]
```diff
@@ -79,15 +79,20 @@ impl Wallet {
...
 
-        let mut secret_manager = self.secret_manager.as_ref().write().await;
-        // Get the current snapshot path if set
-        let new_snapshot_path = if let SecretManager::Stronghold(stronghold) = &mut *secret_manager {
-            stronghold.snapshot_path.clone()
-        } else {
-            PathBuf::from("wallet.stronghold")
-        };
+        // We don't want to overwrite a possible existing wallet
+        // TODO not too sure about this, it used to check the presence of accounts, this is not 100% equivalent
```
Comment (Thoralf-M) [inline sdk/src/wallet/core/operations/stronghold_backup/mod.rs]: I think this is the best way now

File: sdk/src/wallet/core/mod.rs (hunk 1 above)
Comment (DaughterOfMars) [reply sdk/src/wallet/core/mod.rs:100]: All access is immutable. Figured it's better not to have the lock at all in that case. However, if you think there's a valid reason to leave it, let me know.

**Review by Alex6323 — DISMISSED** [review]

File: sdk/src/wallet/core/operations/stronghold_backup/mod.rs [hunk 3]
```diff
@@ -103,16 +107,24 @@ impl Wallet {
...
             if ignore {
                 // TODO: #1279 okay that if both are none we always load the backup values?
-                wallet_data.bip_path != read_bip_path
+                curr_bip_path != read_bip_path
             } else {
                 false
             }
         });
 
+        // TODO: Do we even need to do this, since it should get set when we restore below?
```
Comment (Thoralf-M) [inline sdk/src/wallet/core/operations/stronghold_backup/mod.rs]: I think it's a bit different, it only gets set below `if restore_wallet {` which depends on the bech32 hrp

**Review by Thoralf-M — DISMISSED** [review]

**Review by Alex6323 — DISMISSED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Format

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**